server/src/main/java/org/elasticsearch/action/search/SearchResponse.java

```diff
@@ -118,7 +119,44 @@ public SearchResponse(StreamInput in) throws IOException {
         pointInTimeId = in.readOptionalString();
     }
 
-    public SearchResponse(
+    public static SearchResponse newWithMetrics(
```

Comment:
I wonder if we should move SearchResponse to use a Builder and have the constructor be private.

I have another PR in progress that also needs to add a new param to SearchResponse constructors, so we staring to have an explosion of constructor variants now.

It would be somewhat painful, but I think we could just do a refactoring ticket to create Builders first and then maybe a couple of us could work on converting sections of the codebase to use the constructor like we did for Armin's ref-counting work.

Then we could have one final PR that makes the actual (single) constructor private.

Reply:
It's great that the constructor is protected and could be made package private, already.  It's a great step forward. I did suggest below to perhaps use more self-explanatory names for the different with and without metrics variants. Builders are an option as well, I am curious to see what you folks have in mind in practice. let's discuss this in a follow-up perhaps?

Reply:
For what I'm thinking, look at the SearchResponse.Clusters builder that I added as part of some CCS work.